specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2023-01-01-preview/AlertRuleRecommendations.json

```diff
@@ -0,0 +1,328 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2023-01-01-preview",
+    "title": "Azure Alerts Management Service Resource Provider",
+    "description": "Azure Alerts Management Service provides a single pane of glass of alerts across Azure Monitor."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/{resourceUri}/providers/Microsoft.AlertsManagement/alertRuleRecommendations": {
+      "get": {
+        "tags": [
+          "alertRuleRecommendations"
+        ],
+        "description": "Retrieve alert rule recommendations for a resource.",
+        "operationId": "AlertRuleRecommendations_ListByResource",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceUriParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
```

Comment:
>responses [](http://example.com/codeflow?start=9&length=9)

Adding pagination later, is considered a breaking change. So the recommendation is, if you are sure, your API will never need Pagination, it is fine not to use x-ms-pageable. But if it can be in future, it is better to add it.

Reply:
Adding...